src/app/reskin/hooks/use-reskin-deposit-status.ts

```diff
@@ -0,0 +1,152 @@
+import { useEffect, useMemo, useState } from "react";
+import { useAtomValue } from "jotai";
+import { getEmilyDepositInfo } from "@/util/tx-utils";
+import { bridgeConfigAtom } from "@/util/atoms";
+import {
+  getCurrentBlockHeight,
+  getRawTransaction,
+  getTxRbf,
+} from "@/actions/bitcoinClient";
+import { Cl, PrincipalCV } from "@stacks/transactions";
+import { useEmilyDeposit } from "@/util/use-emily-deposit";
+import { queryClient } from "@/query/client";
+import { useQuery } from "@tanstack/react-query";
+import { DepositStatus } from "@/hooks/use-deposit-status";
+
+// Better logic for better ui
+// you might ask: why not use the same logic as useDepositStatus?
+// cuz we can do better! and we shouldn't modify the original logic
+// since it might break
+export function useReskinDepositStatus(txId?: string) {
+  const { notifyEmily } = useEmilyDeposit();
+  const { RECLAIM_LOCK_TIME, POLLING_INTERVAL } =
+    useAtomValue(bridgeConfigAtom);
+
+  const [transferTxStatus, setStatus] = useState(
+    DepositStatus.PendingConfirmation,
+  );
+
+  const { data: currentBlockHeight } = useQuery({
+    queryKey: ["current-block-height"],
+    queryFn: async () => {
+      return getCurrentBlockHeight();
+    },
+    refetchInterval: () => {
+      const isPending =
+        transferTxStatus !== DepositStatus.Completed &&
+        transferTxStatus !== DepositStatus.Failed;
+      // keep polling until the deposit is completed or failed
+      if (isPending) {
+        return POLLING_INTERVAL;
+      }
+      return false;
+    },
+    enabled: !!txId,
+  });
+
+  const { data: emilyDepositInfo } = useQuery({
+    queryKey: ["emily-deposit-info", txId],
+
+    queryFn: async () => {
+      if (!txId) {
+        throw new Error("txId is required");
+      }
```

Comment:
qq: do we need this, even with the `enabled: !!txId,` below?

Reply:
extra sanity checks basically decoupling the fetch from the query it would break if for example that enabled check was changed to no longer including the txid